Check if libvirt default network is present for ppc64le

mtarsel opened the pull request:
For ppc64le arch, the default network must be present in order to proceed with the snc install.

If default network is not present, the script will exit. The user should install libvirt-daemon-config-network package.

Closes issue #250

cfergeau commented:
/ok-to-test

cfergeau commented:
What is different on ppc64le compared to all others architecture? Why only this arch require the `default` libvirt network?

cfergeau commented:
I believe all arches will try to connect to `192.168.122.1` as this is why we need to enable TCP support for libvirt. So rather than a ppc64le specific test, I'd go with what you initially suggested in #250. This can use `virsh -c qemu:///system net-info default` rather than `virsh net-list`

mtarsel commented:
> Why only this arch require the default libvirt network?

I figured it was only on power since it wasn't a check already in the script. It was related to #227 and I confirmed this on my ppc64le box as well.

I'll redo this commit with a better message. I'll also add a check to make sure the default network is not only present via `net-info` but also Active.

mtarsel commented:
I deleted the previous commit which looks like it closed this PR automatically. I cleaned up the commit and added more comments in this same branch so it's re-opened now. thanks

praveenkumar commented:
/retest

cfergeau commented on snc.sh:
```diff
@@ -74,6 +74,20 @@ function run_preflight_checks() {
+	else
+		echo "Default libvirt network is not available. Exiting now!"
+		exit 1
+	fi
```
Would be simpler as 
```
if ! virsh -c ${LIBVIRT_URI} net-info default &> /dev/null; then
	echo "Default libvirt network is not available. Exiting now!"
	exit 1
fi
echo "default network is available"

#Check if default libvirt network is Active
if [[ $(virsh -c qemu:///system net-info default | awk '{print $2}' | sed '3q;d') == "no" ]]; then
	echo "Default network is not active. Exiting now!"
	exit 1
fi
```
This saves some indentation levels, and the nominal (no error) case is easier to follow.
For the "active" check, I'd just use something like `virsh -c qemu:///system net-info default |grep '^Active: *yes`, this is easier to read, and feels more robust.

mtarsel commented:
rebased and pushed with edits from @cfergeau

praveenkumar commented:
/retest
